snuba/datasets/entities/discover.py

```diff
@@ -251,6 +258,67 @@ def attempt_map(
             return None
 
 
+@dataclass(frozen=True)
+class DefaultNoneFunctionMapper(FunctionCallMapper):
+    """
+    If a function is being called on a column that doesn't exist, or is being
+    called on NULL, change the entire function to be NULL.
+    """
+
+    columns: ColumnSet
+
+    def attempt_map(
+        self,
+        expression: FunctionCall,
```

Comment:
How confident are you hat we can apply this to every function? Which means that `f(non existent column)` will become NULL for every f.

Reply:
Very confident. There is no function that could be applied to nothing that should return something.

Reply:
Actually I think I found an issue:
`....FROM events WHERE isNull(duration) AND ...`
 `isNull(duration)` would have to be converted into `isNull(NULL)` which is true instead it would be converted into `NULL`

There are several clickhouse functions dealing with NULL: https://clickhouse.tech/docs/en/sql-reference/functions/functions-for-nulls/#isnull

Reply:
I think returning `NULL` from `isNull(duration)` is actually correct, because `duration` doesn't exist in the table. Returning True isn't correct, because the data doesn't know if duration is NULL or not. If the user tries to access a column that doesn't exist, they can't get results back. There are no results, NULL or not, to return.